pkg/operator/sync_test.go

```diff
@@ -130,338 +126,170 @@ func TestWaitForDeploymentRollout(t *testing.T) {
 	}
 }
 
-type webhookTestCase struct {
-	testCase             string
-	shouldSync           bool
-	serverOperationError string
-	exisingWebhook       func() *unstructured.Unstructured
-}
-
 func TestSyncValidatingWebhooks(t *testing.T) {
 	defaultConfiguration := mapiv1.NewValidatingWebhookConfiguration()
 
-	testCases := []webhookTestCase{
-		{
-			testCase:   "It should create webhookConfiguration if it does not exsit",
-			shouldSync: true,
-		},
-		{
-			testCase: "It should not update webhookConfiguration if it already exist and is equal expected",
-			exisingWebhook: func() *unstructured.Unstructured {
-				exisingWebhook, _ := runtime.DefaultUnstructuredConverter.ToUnstructured(defaultConfiguration.DeepCopy())
-				return &unstructured.Unstructured{Object: exisingWebhook}
-			},
-			shouldSync: false,
-		},
-		{
-			testCase: "It should not create webhookConfiguration if the server is down",
-			exisingWebhook: func() *unstructured.Unstructured {
-				exisingWebhook, _ := runtime.DefaultUnstructuredConverter.ToUnstructured(defaultConfiguration.DeepCopy())
-				return &unstructured.Unstructured{Object: exisingWebhook}
-			},
-			serverOperationError: "get",
-			shouldSync:           false,
-		},
-		{
-			testCase: "It shouldn't update webhookConfiguration if only caBundle field have changed",
-			exisingWebhook: func() *unstructured.Unstructured {
-				webhook := defaultConfiguration.DeepCopy()
-				webhook.Webhooks[0].ClientConfig.CABundle = []byte("test")
-				exisingWebhook, _ := runtime.DefaultUnstructuredConverter.ToUnstructured(webhook.DeepCopy())
-				return &unstructured.Unstructured{Object: exisingWebhook}
-			},
-			shouldSync: false,
-		},
-		{
-			testCase: "It should update webhookConfiguration if some of their webhooks differ",
-			exisingWebhook: func() *unstructured.Unstructured {
-				webhook := defaultConfiguration.DeepCopy()
-				webhook.Webhooks[0].Name = "test"
-				exisingWebhook, _ := runtime.DefaultUnstructuredConverter.ToUnstructured(webhook)
-				return &unstructured.Unstructured{Object: exisingWebhook}
-			},
-			shouldSync: true,
-		},
-		{
-			testCase: "It should not update webhookConfiguration if some of their webhooks differ, but the server is down",
-			exisingWebhook: func() *unstructured.Unstructured {
-				webhook := defaultConfiguration.DeepCopy()
-				webhook.Webhooks[0].Name = "test"
-				exisingWebhook, _ := runtime.DefaultUnstructuredConverter.ToUnstructured(webhook)
-				return &unstructured.Unstructured{Object: exisingWebhook}
-			},
-			serverOperationError: "update",
-			shouldSync:           false,
-		},
+	withCABundle := defaultConfiguration.DeepCopy()
+	for i, webhook := range withCABundle.Webhooks {
+		webhook.ClientConfig.CABundle = []byte("test")
+		webhook.TimeoutSeconds = pointer.Int32Ptr(10)
```

Comment:
I don't think this should be there. If someone changes default timeout onto `10` seconds, we should bring it back to the one we want to see there (even if it is equal, but we don't set it, we need to apply)

Reply:
Does the timeout affect the operation of the webhook? If you think having the timeout as a particular value is critical to the operation of the webhooks then we should be forcing it. I'm currently leaning towards we don't really care about this value (hence using it in the tests), but am open to discussing it if you feel strongly about this.